interface-elements-for-desktop/articles/report-designer/report-designer-for-winforms/create-popular-reports/create-a-hierarchical-report.md

```diff
@@ -0,0 +1,104 @@
+
+2. [Bind the report](../bind-to-data.md) to a required data source.
+
+    The following image demonstrates an empty report bound to an [ObjectDataSource](..\bind-to-data\bind-a-report-to-an-object-data-source.md).
+
+    ![HierarchicalReport-SelectFields](../../../../images/eurd-HieararchicalReport-CreateReport.png)
+
+    Each record in this data source includes the "parent ID" field that defines the parent-child relationship and thus builds the hierarchy.
+
+3. Arrange controls on the report.
+
+    - Add the [Report Header](..\introduction-to-banded-reports.md) and [Page Header](..\introduction-to-banded-reports.md) bands (see the **Manage Report Bands | Add Bands** section in the [Introduction to Banded Reports](..\introduction-to-banded-reports.md) document for details)
+    - Add [data-bound labels](..\use-report-elements\use-basic-report-controls\label.md) to the **Detail** band.
+
+    ![HierarchicalReport-ArrangeControls](../../../../images/eurd-win-HierarchicalReport-ArrangeControls.png)
+
+    Switch to the [Preview](..\preview-print-and-export-reports.md) tab to see an intermediate result.
+
+    ![HierarchicalReport-ArrangeControls-Result](../../../../images/eurd-win-HierarchicalReport-ArrangeControls-Result.png)
+
+4. Specify the Detail band's **Hierarchy Print Options** property.
+
+    ![HierarchicalReport-Set-HierarchyPrintOptions](../../../../images/eurd-win-HierarchicalReport-Set-HierarchyPrintOptions.png)
+
+    Set the following options:
+
+    - **Key Field Name** and **Parent Field Name**, or **Child List Field Name**  
+    Set the **Key Field Name** and **Parent Field Name** properties if your report's data have the Id-ParentID related fields.  
+    Set the **Child List Field Name** property if your report's data is recursive. Assign the collection of child objects (records) if they have the same type as the parent objects (records).
+    
+    - **Indent**   
+    Specify the child level node offset.
+
+    - **Keep Together with First Child**  
+    Specify whether to print a parent node together with its first child node on the next page, if these nodes do not fit at the end of a page.
+
+    ![HierarchicalReport-HierarchyPrintOptions-Result](../../../../images/eurd-win-HierarchicalReport-HierarchyPrintOptions-Result.png)
+
+    As you can see in the image above, the **Detail** band that contains child rows is printed with the specified indent. However, the row (the sum of the label widths) do not fit the page now.
+
+5. Align labels.
+
+    - Anchor the first data-bound label to the Detail band's left and right edges. Set the label's **Anchor Horizontally** property to **Both**.
+
+        ![HierarchicalReport-AnchorHorizontally-both](../../../../images/eurd-win-hierarchicalreports-anchorhorizontally-both.png)
+
+    - Anchor the rest data-bound labels to the right edge of the Detail band (their container). Set their **Anchor Horizontal** property to **Rignt**.    
```

Comment:
Anchor the rest of the data-bound labels to the right edge of the Detail band (their container). Set their **Anchor Horizontal** property to **Right**.